src/widgets/settingspages/HighlightingPage.cpp

```diff
@@ -130,6 +143,53 @@ HighlightingPage::HighlightingPage()
                                  });
             }
 
+            auto badgeHighlights = tabs.appendTab(new QVBoxLayout, "Badges");
+            {
+                badgeHighlights.emplace<QLabel>(
+                    "Play notification sounds and highlight messages based on "
+                    "user badges.");
+                auto view = badgeHighlights
+                                .emplace<EditableModelView>(
+                                    (new BadgeHighlightModel(nullptr))
+                                        ->initialized(
+                                            &getSettings()->highlightedBadges))
+                                .getElement();
+                view->setTitles({"Name", "Flash\ntaskbar", "Play\nsound",
+                                 "Custom\nsound", "Color"});
+                view->getTableView()->horizontalHeader()->setSectionResizeMode(
+                    QHeaderView::Fixed);
+                view->getTableView()->horizontalHeader()->setSectionResizeMode(
+                    0, QHeaderView::Stretch);
+
+                // fourtf: make class extrend BaseWidget and add this to
+                // dpiChanged
+                QTimer::singleShot(1, [view] {
+                    view->getTableView()->resizeColumnsToContents();
+                    view->getTableView()->setColumnWidth(0, 200);
+                });
+
+                view->addButtonPressed.connect([this] {
+                    auto d = std::make_shared<BadgePickerDialog>(
+                        availableBadges, this);
+
+                    d->setWindowTitle("Choose badge");
+                    if (d->exec() == QDialog::Accepted)
+                    {
+                        auto s = d->getSelection();
+                        getSettings()->highlightedBadges.append(
+                            HighlightBadge{s->badgeName(), s->badgeVersion(),
+                                           s->displayName(), false, false, "",
+                                           ColorProvider::instance().color(
+                                               ColorType::SelfHighlight)});
+                    }
```

Comment:
```suggestion
                    if (d->exec() != QDialog::Accepted)
                    {
                        return;
                    }
                    auto s = d->getSelection();
                    if (!s)
                    {
                        return;
                    }
                    getSettings()->highlightedBadges.append(
                        HighlightBadge{s->badgeName(), s->badgeVersion(),
                                       s->displayName(), false, false, "",
                                       ColorProvider::instance().color(
                                           ColorType::SelfHighlight)});
```

Reply:
I think it's better to stick with the convention already used:
https://github.com/Chatterino/chatterino2/blob/6d5ba0c442681fb30afd01f787a10cd3922c10aa/src/widgets/dialogs/SelectChannelDialog.cpp#L153-L156

https://github.com/Chatterino/chatterino2/blob/6d5ba0c442681fb30afd01f787a10cd3922c10aa/src/widgets/helper/NotebookTab.cpp#L77-L81

But I did add the optional checking (4ac85770)
```c++
if (!s)
{
    return;
}
```